balancer_res.go

```diff
@@ -0,0 +1,106 @@
+package metafora
+
+import (
+	"fmt"
+	"time"
+)
+
+// ResourceReporter is required by the ResourceBalancer to read the resource
+// being used for balancing.
+type ResourceReporter interface {
+	// Used returns the amount of a resource used and the total amount of that
+	// resource.
+	Used() (used uint64, total uint64)
+
+	// String returns the plural name of the unit resources are reported in.
+	String() string
+}
+
+// ResourceBalancer is a balancer implemntation which uses two thresholds to
+// limit claiming and rebalance work based upon a resource reported by a
+// ResourceReporter. When the claim threshold is exceeded, no new work will be
+// claimed. When the release threshold is exceeded work will be released until
+// below that threshold. The claim threshold must be less than the release
+// threshold (otherwise claims would continue just to have the work
+// rebalanced.)
+//
+// Even below the claim limit, claims are delayed by the percent of resources
+// used (in milliseconds) to give less loaded nodes a claim advantage.
+//
+// The balancer releases the oldest tasks first (skipping those who are already
+// stopping) to try to prevent rebalancing the same tasks repeatedly within a
+// cluster.
+type ResourceBalancer struct {
```

Comment:
Added that to the NewResourceBalancer functions docs since that's where they're set. Think that's good enough?